nemo-benches/benches/output.rs

```diff
@@ -0,0 +1,138 @@
+        b.iter_batched(
+            || physical_constants.clone(),
+            |output| {
+                for _ in output.into_iter() {}
+            },
+            BatchSize::SmallInput,
+        )
+    });
+
+    physical_output.finish();
+
+    let mut logical_output = c.benchmark_group("output_logical_values");
+
+    logical_output.bench_function("logical_strings", |b| {
+        b.iter_batched(
+            || physical_strings.clone(),
+            |output| {
+                for _ in PrimitiveType::String.primitive_logical_value_iterator(
+                    DataValueIteratorT::String(Box::new(output.into_iter())),
+                ) {}
+            },
+            BatchSize::SmallInput,
+        )
+    });
+
+    logical_output.bench_function("strings_in_logical_any", |b| {
+        b.iter_batched(
+            || physical_strings.clone(),
+            |output| {
+                for _ in PrimitiveType::Any.primitive_logical_value_iterator(
+                    DataValueIteratorT::String(Box::new(output.into_iter())),
+                ) {}
+            },
+            BatchSize::SmallInput,
+        )
+    });
+
+    logical_output.bench_function("constants_in_logical_any", |b| {
+        b.iter_batched(
+            || physical_constants.clone(),
+            |output| {
+                for _ in PrimitiveType::Any.primitive_logical_value_iterator(
+                    DataValueIteratorT::String(Box::new(output.into_iter())),
+                ) {}
+            },
+            BatchSize::SmallInput,
+        )
+    });
+
+    logical_output.finish();
+
+    let mut serialized_output = c.benchmark_group("output_serialized_values");
+
+    serialized_output.bench_function("logical_strings", |b| {
+        b.iter_batched(
+            || physical_strings.clone(),
+            |output| {
+                for _ in PrimitiveType::String
+                    .serialize_output(DataValueIteratorT::String(Box::new(output.into_iter())))
+                {
+                }
+            },
+            BatchSize::SmallInput,
+        )
+    });
+
+    serialized_output.bench_function("strings_in_logical_any", |b| {
+        b.iter_batched(
+            || physical_strings.clone(),
+            |output| {
+                for _ in PrimitiveType::Any
+                    .serialize_output(DataValueIteratorT::String(Box::new(output.into_iter())))
+                {
+                }
+            },
+            BatchSize::SmallInput,
+        )
+    });
+
+    serialized_output.bench_function("constants_in_logical_any", |b| {
+        b.iter_batched(
+            || physical_constants.clone(),
+            |output| {
+                for _ in PrimitiveType::Any
+                    .serialize_output(DataValueIteratorT::String(Box::new(output.into_iter())))
+                {
+                }
+            },
+            BatchSize::SmallInput,
+        )
+    });
+
+    serialized_output.finish();
+}
```

Comment:
Interestingly, out of these two the latter is noticeably faster. I did not investigate this too much but I would have expected them to have more or less the same performance.